code/script.js

```diff
@@ -222,104 +226,195 @@ const generateBoard = () => {
   })
 }
 
+
 // Randomly select a person from the characters array and set as the value of the variable called secret
 const setSecret = () => {
-  secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
+ secret  = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
 }
 
+
 // This function to start (and restart) the game
 const start = () => {
+  document.getElementById("winOrLose").style.display = "none";
+
   // Here we're setting charactersInPlay array to be all the characters to start with
   charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
+
+  // The generateBoard function is called that draws the gameboard
+  generateBoard()
+  
+  // setSecret selects a secret character randomly from the CHARACTERS array
+  setSecret() 
+
+  // setting the currentQuestion object when you select something in the dropdown
+  selectQuestion()
 }
 
 // setting the currentQuestion object when you select something in the dropdown
-const selectQuestion = () => {
+  const selectQuestion = () => {
   const category = questions.options[questions.selectedIndex].parentNode.label
 
+  // Variable that stores the actual value of the question we've selected.
+
+  const actualValue = questions.value
+
+
   // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
 
   currentQuestion = {
-    category: category,
-    // value: value
+  category: category,
+  value: actualValue,
+    }
   }
-}
+
 
 // This function should be invoked when you click on 'Find Out' button.
+// It'll filter the characters array and redraw the game board.
+
+
 const checkQuestion = () => {
-  const { category, value } = currentQuestion
+  const { category, value } = currentQuestion;
+
 
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
   if (category === 'hair' || category === 'eyes') {
+    if (secret[category] === value) {
```

Comment:
Nice solution for the filters! looks really clean =)

Reply:
Well thank YOU for helping me with this section ;) :D